### PR TITLE
Combat improvements

### PR DESCRIPTION
**Pull request by Malthusis**
This was getting a bit bloated, need to merge this up.
This adds a few improvements to the combat menu/system. Included is:

- Custom button assets and some basic background for future character sprites. Add animated effects to buttons on hover.
- Make HP bars look cooler
- Added "Explore" functionality to keep looking for enemies after you've killed one.
- Added a simple keyboard shortcut to attack, more on the way.
- Created SP, a resource for skills.
- Created a SkillsMenu to use skills in. Currently in progress, next PR will fill this out.
- Clean up a bunch of minor CSS stuff.

Up next:
- Actually populate the skills window with skills.
- Hook up some upgrades to stitch it properly into the game.

Before:
image

After:
Combat

### CODE REVIEW

File: src/components/CombatPanel.vue [hunk 1]
```diff
@@ -33,21 +37,20 @@
 
 
         <div class="stats_flex_container">
+            <SkillsMenu :open="skillsMenuActive"></SkillsMenu>
```
Comment (cm97878) [inline src/components/CombatPanel.vue:40]: This could probably just be `<SkillsMenu :open="xyz" />`

File: src/stores/player.ts [hunk 2]
```diff
@@ -115,6 +118,24 @@ export const usePlayer = defineStore('player', () => {
 
     const getHPRegen = computed(() => baseStats.value.hpRegen)
     const getEnergyRegen = computed(() => baseStats.value.energyRegen)
+    const soulOrbs = computed(() => {
+        let orbArray = [] as boolean[];
+        let i: number = 0;
+        let orbNum: number = 0;
+        debugger;
```
Comment (cm97878) [inline src/stores/player.ts:125]: i beg of you, remove your debugger lines when you push, they drive me nuts

**Comment by cm97878** [comment]
looks great!

File: src/stores/player.ts (hunk 2 above)
Comment (Malthusis) [reply src/stores/player.ts:125]: I am blind, didn't notice it. My bad 😅

**Pull request merged**